### ISSUE TITLE
Optimize NLP Model Training: Current Duration of 4 Hours Needs Improvement

### ISSUE THREAD

**Issue by DayaYagu777** [DayaYagu777/DataScienceGirl#1]
## Current Behavior
The NLP model training process currently takes approximately 4 hours to complete with the following configuration:

```python
# Training Arguments Configuration
learning_rate = 3.6e-5
per_device_train_batch_size = 32
per_device_eval_batch_size = 32
num_train_epochs = 3
weight_decay = 0.01
warmup_steps = 600
fp16 = False
```

## Performance Impact
- Training duration: ~4 hours
- Current batch sizes: 32 for both training and evaluation
- No FP16 optimization enabled

## Potential Optimization Areas

1. **Mixed Precision Training**
   - Current setting: `fp16 = False`
   - Consider enabling FP16 training to improve speed and memory usage

2. **Batch Size Optimization**
   - Current batch sizes: 32
   - Could potentially be increased if memory allows
   - Consider gradient accumulation if larger batches are needed

3. **Hardware Utilization**
   - Need to verify current GPU utilization
   - Check memory usage during training
   - Consider distributed training if available

## Technical Details
```python
from transformers import TrainingArguments

training_args = TrainingArguments(
    output_dir="profner_classifier_results",
    per_device_train_batch_size=32,
    per_device_eval_batch_size=32,
    learning_rate=3.6e-5,
    num_train_epochs=3,
    weight_decay=0.01,
    warmup_steps=600,
    fp16=False,
    eval_strategy="epoch",
    save_strategy="epoch",
    save_total_limit=2,
    load_best_model_at_end=True
)
```

## Suggested Investigations
1. Profile memory usage during training
2. Monitor GPU utilization
3. Test with FP16 enabled
4. Experiment with larger batch sizes
5. Consider gradient accumulation steps
6. Evaluate hardware resource utilization

## Success Criteria
- Reduce training time by at least 50% without significant impact on model performance
- Maintain or improve the current F1 score (metric_for_best_model)
